src/lamp_py/ingestion/compress_gtfs/gtfs_to_parquet.py

```diff
@@ -0,0 +1,321 @@
+            for batch in filter_ds.to_batches(batch_size=batch_size):
+                writer.write_batch(batch)
+
+        # over-write export_path file with merged dataset
+        combined_ds = pd.dataset((filter_pq_path, new_pq_path))
+        with pq.ParquetWriter(export_path, schema=pq_schema) as writer:
+            for batch in combined_ds.to_batches(batch_size=batch_size):
+                writer.write_batch(batch)
+
+
+def compress_gtfs_file(
+    gtfs_table_file: str, schedule_details: ScheduleDetails
+) -> None:
+    """
+    compress an indivdual gtfs_table_file (ie. stop_times.txt) into yearly parquet
+    partitioned parquet file(s)
+
+    yearly partition is based on ScheduleDetals.active_from_int value (1 day after published_dt)
+
+    will perform 1 of 3 operations:
+    1.  if a yearly parquet file already exists, perform differential
+        merge operation on existing parquet file with gtfs_table_file
+
+    2.  yearly parquet file does not exist, but previous year file does exist:
+        - update previous year file with "gtfs_end_date" values going to end of previous year
+        - create new yearly partition for current year (with merged records from previous year)
+
+    3.  no parition files exist (current or previous year), create new partition
+        file for current year (process initialization)
+
+    :param gtfs_table_file: (ie. stop_times.txt)
+    :param schedule_details: data required for schedule compression operation
+    """
+    partition_year = int(str(schedule_details.active_from_int)[:4])
+
+    gtfs_table = gtfs_table_file.replace(".txt", "")
+
+    export_path = os.path.join(
+        schedule_details.tmp_folder,
+        f"{partition_year}",
+        f"{gtfs_table}.parquet",
+    )
+    last_export_path = os.path.join(
+        schedule_details.tmp_folder,
+        f"{partition_year-1}",
+        f"{gtfs_table}.parquet",
+    )
+
+    new_frame = schedule_details.gtfs_to_frame(gtfs_table_file)
+
+    if os.path.exists(export_path):
+        #
+        # regular merge operation (with export_path)
+        #
+        old_records, same_records, new_records = frame_parquet_diffs(
+            new_frame=new_frame,
+            pq_path=export_path,
+            gtfs_table_file=gtfs_table_file,
+            filter_date=schedule_details.active_from_int,
+        )
+
+        # "gtfs_end_date":
+        #   (same or new records) set to schedule_details.active_to_int
+        #   (old records) set to schedule_details.published_int (day before active_to_int)
+        same_records = same_records.with_columns(
+            pl.lit(schedule_details.active_to_int).alias("gtfs_end_date")
+        )
+        old_records = old_records.with_columns(
+            pl.lit(schedule_details.published_int).alias("gtfs_end_date")
+        )
+
+        merge_records = pl.concat(
+            (old_records, same_records, new_records),
+            how="diagonal",
+        )
```

Comment:
tuples are technically lower overhead and generally meant to be used when creating an immutable list-type object, so that seems to apply in these cases

The `diagonal` merge is more forgiving with the structure of the data-frames that are passed into the `concat` operation. I believe without `diagonal` the frames being combined have to all be the exact same format, same ordering of columns and everything. Removing `diagonal` from these calls does cause the test to fail.